components/compass/label/label.tsx

```diff
@@ -2,18 +2,24 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
+import classNames from 'classnames';
 
-import './label.scss';
-
-export default class Label extends React.PureComponent {
-    render() {
-        return (
-            <span
-                className='Label Label___standard'
-                title='Lorem ipsum dolor sit amet, consectetur adipiscing.'
-            >
-                {'Lorem ipsum dolor sit amet, consectetur adipiscing.'}
-            </span>
-        );
-    }
+interface Props {
+    children: React.ReactNode;
+    size: string;
+    wrap?: boolean;
+    onMouseOver?: () => void;
 }
+
+const Label: React.FC<Props> = ({children, size, wrap, onMouseOver}: Props) => {
+    return (
+        <span
+            className={classNames([`Label Label___${size}`, {Label___wrap: wrap}])}
+            onMouseOver={onMouseOver}
```

Comment:
I was thinking about taking the opportunity to be smart about passing this event up to the parent. We really only need a tooltip if the label is truncated due to space restrictions so I was thinking we only need to tell the parent the user is currently moused over the label if the label's text is truncated. There's no reason to have a tooltip otherwise and requiring the parent to determine if a label is truncated does't feel like the right approach to me. The event function name might need to be customized to make sense if we take this approach.

We also need to consider how to let the parent know the user has moused away so the parent can appropriately dismiss any open tooltips.

Reply:
So we can have the 'truncated' prop passed to the parent? I think this creates a two-way flow which is an anti-pattern. Or did I misunderstand the suggestion? I think in the function we can pass 'truncated = true' when this happens render the tooltip from the parent. 

```const onMouseOver = (truncated) => {
         if (truncated) {
             ...render toolTip
         }
    }
```